### PR TITLE
docs: document usage

### PR DESCRIPTION
**Pull request by HEPOSHEIKKI**
Improved the README and added the APISpec and themes documentation in the `docs/` directory.

This is my first time writing documentation so feel free to correct any mistakes or improve the formatting.

### CODE REVIEW

File: docs/Themes.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
+# Themes
+
+Themes are used to customise the colors of the card. github-readme-testaustime ships with a whopping 60 themes to choose from.
```
Comment (Chicken) [inline docs/Themes.md:3]: We could just link the upstream's theme listing <https://github.com/anuraghazra/github-readme-stats/blob/678fa1141676fc8fb75e2a322e4297323900e330/themes/README.md>

File: docs/APISpec.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
+# APISpec
+
+## Usage
+
+All requests for a card are made to `/api/testaustime` and parameters are passed as query strings.
+
+## Endpoints
+The API provides 17 routes:
```
Comment (Chicken) [inline docs/APISpec.md]: Calling these endpoints/routes is wrong, it's really just a single endpoint/route with query string arguments

File: docs/APISpec.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
+# APISpec
+
+## Usage
```
Comment (Chicken) [inline docs/APISpec.md:3]: We could again link to upstream docs and only mention differences
<https://github.com/anuraghazra/github-readme-stats/tree/678fa1141676fc8fb75e2a322e4297323900e330#common-options>
<https://github.com/anuraghazra/github-readme-stats/tree/678fa1141676fc8fb75e2a322e4297323900e330#wakatime-card-exclusive-options>
This isn't really necessary though, this is fine as is.

**Comment by HEPOSHEIKKI** [comment]
This probably closes #1 as well

File: docs/APISpec.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+### coloring <a name="coloring"></a>
+Due to all of them using the same syntax, this part covers the following functions:
+
+    title_color
+    bg_color
+    text_color
+    icon_color
+    border_color
+
+Used to set the color of the respective element in hex format. These values will override the current theme if one is specified.
```
Comment (Chicken) [inline docs/APISpec.md:154]: Apparently the `bg_color` can be a comma seperated list for gradients

**Review by Chicken — CHANGES_REQUESTED** [review]

File: docs/Themes.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+
+<details>
+    <summary>default</summary>
+  
+![image](https://github.com/HEPOSHEIKKI/github-readme-testaustime/assets/51783206/6a726abe-5508-412a-954d-e1823dbc630f)
```
Comment (Chicken) [inline docs/Themes.md:8]: Oh yeah, these image urls are a bit weird, not sure how they will work when merged and fork is deleted.

**Review by Chicken — APPROVED** [review]

**Pull request merged**